nengo/connection.py

```diff
@@ -117,25 +117,70 @@ def validate(self, conn, distorarray):
 class ConnectionFunctionParam(FunctionParam):
     """Connection-specific validation for functions."""
 
+    def __set__(self, conn, function):
+        if function is None:
+            function_info = self.Info(function=None, size=None)
+        elif is_array_like(function):
+            array = np.array(function, copy=False, dtype=np.float64)
+            self.validate_array(conn, array)
+            function_info = self.Info(function=array, size=array.shape[1])
+        elif callable(function):
+            function_info = self.Info(function=function,
+                                      size=self.determine_size(conn, function))
+            self.validate_callable(conn, function_info)
+        else:
+            raise ValidationError("Invalid connection function type %r "
+                                  "(must be callable or array-like)"
+                                  % function.__class__.__name__,
```

Comment:
I was assuming that everything has a `__class__` attribute, but I don't think that's quite true. Better to use `type()` instead? There are probably other places we make this assumption, too (e.g. node outputs).